src/util/hostname.cpp

```diff
@@ -0,0 +1,30 @@
+#include <string>
+
+#include "hostname.hpp"
+
+#ifdef __linux__
+extern "C" {
+    #include <unistd.h>
+}
+#endif
+
+namespace nest {
+namespace mc {
+namespace util {
+
+#ifdef __linux__
+std::string hostname() {
+    char name[128];
+    auto result = gethostname(name, sizeof(name));
+    return result? "unknown": name;
+}
+#else
+std::string hostname() {
+    return "unknown";
```

Comment:
It's perverse, but someone could name their machine 'unknown'. Perhaps have this function return an empty string or other form of marked error response, and have the caller handle the user-visible signal for that error?